### PR TITLE
LCH-2395: Add raw search criteria to a proprietary HTTP header before relaying the request

### PR DESCRIPTION
**Pull request by c0d39uru**
The aim is to add raw search criteria to a proprietary request header right before it gets sent to the server.
I tried to clean up and simplify former implementation in favor of more maintainable code. Also, I wrote unit test to cover all cases that came to my mind. Please feel free to point out any edge case that this implementation doesn't cover. 

You can run the unit test using the following command:
./vendor/phpunit/phpunit/phpunit -c phpunit.xml test/SearchCriteriaTest.php

### CODE REVIEW

File: test/SearchCriteriaTest.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Acquia\ContentHubClient\test;
+
+use PHPUnit\Framework\TestCase;
+use Acquia\ContentHubClient\SearchCriteria\Transformer;
+
+
```
Comment (saxumVermes) [inline test/SearchCriteriaTest.php]: Accidental extra blank line.

File: test/SearchCriteriaTest.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      'sorting' => '',
+      'version' => 'some-version',
+      'languages' =>
+        [
+          0 => 'some-language-1',
+          1 => 'some-language-2',
+        ],
+    ];
+  }
+  public function test_result_when_all_keys_are_present_in_data(): void {
```
Comment (saxumVermes) [inline test/SearchCriteriaTest.php]: This would be the place of that extra blank line.

File: test/SearchCriteriaTest.php [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    $data = $this->data;
+    $expected = $this->expected;
+
+    self::assertEquals(Transformer::arrayToSearchCriteriaArray($data), $expected);
+  }
+
+  public function test_missing_keys_get_default_values(): void {
```
Comment (saxumVermes) [inline test/SearchCriteriaTest.php]: Why does this differ in terms of code style from the other tests?

**Comment by saxumVermes** [comment]
Why don't we use the phpunit annotations for more accurate test coverage report?

**Comment by saxumVermes** [comment]
addSearchCriteriaHeader method hasn't been included in the tests. Could you provide a coverage for it?

File: test/SearchCriteriaTest.php (hunk 3 above)
Comment (c0d39uru) [reply test/SearchCriteriaTest.php]: Could you please elaborate on this?

**Comment by looking-4-justice** [comment]
> addSearchCriteriaHeader

**Comment by c0d39uru** [comment]
> addSearchCriteriaHeader method hasn't been included in the tests. Could you provide a coverage for it?

There are lots of methods that do not have test coverage at all. AFAIK, we are gonna either rewrite this library from scratch or have dedicated sprint(s) to adding unit tests to it. I believe that's the right time to cover this.

**Comment by looking-4-justice** [comment]
I accidentally closed this PR

**Comment by c0d39uru** [comment]
> Why don't we use the phpunit annotations for more accurate test coverage report?

Not sure if I completely understand this. Please elaborate on your comment a bit.

File: test/SearchCriteriaTest.php (hunk 3 above)
Comment (saxumVermes) [reply test/SearchCriteriaTest.php]: I mean here you use snake cased method names, while in other test classes camel case is used.

**Comment by saxumVermes** [comment]
> > Why don't we use the phpunit annotations for more accurate test coverage report?
> 
> Not sure if I completely understand this. Please elaborate on your comment a bit.

Sure, I was thinking about this: https://phpunit.readthedocs.io/en/8.2/annotations.html

**Review by saxumVermes — APPROVED** [review]
